### ISSUE TITLE
pact-jvm-provider-maven: time-out settings for state change?

### ISSUE THREAD

**Issue by ndelacrz** [DiUS/pact-jvm#490]
We're using the pact-jvm-provider-maven and we are using a "provider state change service" to perform the set-up and tear-down for each test.
However, from time-to-time, the execution of the pact-jvm-provider-maven's verify is getting HTTP 504 when trying to send a POST to the "provider state change service", most probably due to our slow network. 
So what we're thinking is to somehow increase the timeout period (amount of time the pact-jvm-provider-maven will wait for the "provider state change service" to respond)
Is there such a parameter?

**Comment by uglyog** [comment]
Good idea, both the connection and socket timeout values should be able to be changed on the HTTP client. It just needs to be configurable via the maven plugin.
